tasks/sync_pull.py

```diff
@@ -166,8 +168,10 @@ async def run_async_within_lock(
         commits = None
         db_session.commit()
         try:
-            commits = await repository_service.get_pull_request_commits(pull.pullid)
-            base_ancestors_tree = await repository_service.get_ancestors_tree(
+            commits = async_to_sync(repository_service.get_pull_request_commits)(
+                pull.pullid
+            )
+            base_ancestors_tree = async_to_sync(repository_service.get_ancestors_tree)(
```

Comment:
this is an example of where asyncio could be used successfully:
```
commits_future = repository_service.get_pull_request_commits(pull.pullid)
ancestor_tree_future = repository_service.get_ancestors_tree(enriched_pull.provider_pull["base"]["branch"])
commits, base_ancestors_tree = async_to_sync(asyncio.gather)(commits_future, ancestor_tree_future)
```
i didn't though because doing that makes a test fail. as-written, the commits call raises an exception and the ancestors tree call is never run. they both run if done as above, but this test passes a `set()` instead of a `dict()` for `provider_pull` so the test fails when trying to subscript it. besides the failure, it's a slight difference in behavior that doesn't belong here